lib/daemon_runner/session.rb

```diff
@@ -107,6 +114,18 @@ def destroy!
       Diplomat::Session.destroy(id)
     end
 
+    # Verify wheather the session exists after a period of time
+    def verify_session
+      logger.info(" - Wait until Consul session #{id} exists")
+      Timeout.timeout(2) do
+        sleep 1 if session_exist?
```

Comment:
I have found the `/v1/session/list` api to be very flakey, sometimes session will stick sometimes they will not.  An immediate get to `/v1/session/list` will almost always return the session, but after some short period of time(less than the TTL) the session will be gone.

Yes I know this is gross, do you have suggestions?

> 2 seconds is a pretty short period of time

I didn't want to block the session create process for longer that necessary.